programs/sbf/rust/invoke/src/lib.rs

```diff
@@ -1456,19 +1458,38 @@ fn process_instruction<'a>(
             heap[8..pos].fill(42);
 
             // Check that the stack is zeroed too.
-            //
+            let stack = unsafe {
+                slice::from_raw_parts_mut(
+                    MM_STACK_START as *mut u8,
+                    MAX_CALL_DEPTH * STACK_FRAME_SIZE,
+                )
+            };
+
+            #[cfg(not(feature = "dynamic-frames"))]
             // We don't know in which frame we are now, so we skip a few (10) frames at the start
             // which might have been used by the current call stack. We check that the memory for
             // the 10..MAX_CALL_DEPTH frames is zeroed. Then we write a sentinel value, and in the
             // next nested invocation check that it's been zeroed.
-            let stack =
-                unsafe { slice::from_raw_parts_mut(MM_STACK_START as *mut u8, 0x100000000) };
+            //
+            // When we don't have dynamic stack frames, the stack grows from lower addresses
+            // to higher addresses, so we compare accordingly.
             for i in 10..MAX_CALL_DEPTH {
                 let stack = &mut stack[i * STACK_FRAME_SIZE..][..STACK_FRAME_SIZE];
                 assert!(stack == &ZEROS[..STACK_FRAME_SIZE], "stack not zeroed");
                 stack.fill(42);
             }
 
+            #[cfg(feature = "dynamic-frames")]
```

Comment:
Numbers here should not be hard coded but calculated from the constants (`MAX_CALL_DEPTH * STACK_FRAME_SIZE`).

Reply:
I can make a guesstimate of `(MAX_CALL_DEPTH - 2) * STACK_FRAM_SIZE` for here, but the truth is that estimating the number of zeroed bytes with `STACK_FRAME_SIZE = 4096` is more intricate now, since stack frames have no predefined size.